### PR TITLE
Add MSVC build to CI

### PR DESCRIPTION
**Pull request by eustas**

### CODE REVIEW

**Comment by eustas** [comment]
Now it compiles with MSVC, but gives gazillion warnings / errors. Do not land, working on that.

**Comment by fabiangreffrath** [comment]
Yes, one of them is this:

```
D:\a\faad2\faad2\frontend\main.c(51,10): fatal  error C1083: Cannot open include file: 'getopt.h': No such file or directory [D:\a\faad2\faad2\build\faad_cli.vcxproj]
```

https://github.com/knik0/faad2/actions/runs/5378946489/jobs/9759462237?pr=159#step:3:16038

**Comment by eustas** [comment]
Yup. Most likely we have to fix it inside CMake buildfile (add to sources). In progress.

**Comment by fabiangreffrath** [comment]
MSVC is annoying AF with its warnings. Especially because none of them pushes you to write *better* code, just *different* code.

**Comment by eustas** [comment]
Sorry for spamming with builds, it is easier for me to move in small steps...

**Comment by fabiangreffrath** [comment]
> Sorry for spamming with builds, it is easier for me to move in small steps...

Yes, of course, no problem!

**Comment by fabiangreffrath** [comment]
Wow, only ~500 warnings left. Only actual error seems to be the getopt issue.

File: CMakeLists.txt [hunk 1]
```diff
@@ -175,11 +188,19 @@ transform_pc_file("libfaad/faad2.pc.in" "${CMAKE_CURRENT_BINARY_DIR}/faad2.pc" "
...
+    frontend
+  )
+else()
+  set_target_properties(faad_cli PROPERTIES OUTPUT_NAME faad)
+endif()
 
 # Installation
 
-if(NOT FAAD_BUNDLED_MODE)
+if(NOT FAAD_BUNDLED_MODE AND NOT MSVC)
```
Comment (fabiangreffrath) [inline CMakeLists.txt:203]: Built! 🎉

Comment (fabiangreffrath) [reply CMakeLists.txt:203]: There are some new GCC/Clang warnings, though, so if you could iron them out as well... 😉

Comment (eustas) [reply CMakeLists.txt:203]: Sure. Also I would like to run some manual tests to see nothing is broken =)

Comment (eustas) [reply CMakeLists.txt:203]: Ready for review

**Review by fabiangreffrath — APPROVED** [review]
Impressive, thank you very much!

**Pull request merged**